tests/client/components/carousel.spec.js

```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
+      const wrapper = mount(Carousel);
+
+      expect(wrapper.vm.slideCount).toBe(0);
+
+      expect(wrapper).toMatchSnapshot();
+    });
+
+    it('should register 3 slides when 3 slides are added to the slots', done => {
+      const wrapper = mount(Carousel, {
+        slots: {
+          default: [Slide, Slide, Slide]
+        }
+      });
+
+      expect(wrapper.vm.slideCount).toBe(3);
+
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
   });
 
-  it('should unmount successfully', () => {
-    const wrapper = mount(Carousel);
+  describe('Adjustable height easing', () => {
+    it('should not set adjustable height transition easing', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          adjustableHeight: false,
+          adjustableHeightEasing: 'linear',
+          speed: 0
+        }
+      });
 
-    const carouselInstance = wrapper.vm;
-    carouselInstance.$destroy();
-    expect(carouselInstance._isDestroyed).toBeTruthy();
+      expect(wrapper.vm.transitionStyle).toBe('0s ease transform');
 
-    expect(wrapper).toMatchSnapshot();
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
+
+    it('should set adjustable height transition easing', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          adjustableHeight: true,
+          adjustableHeightEasing: 'linear',
+          speed: 0
+        }
+      });
+
+      expect(wrapper.vm.transitionStyle).toBe('0s ease transform, height 0s linear');
+
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
   });
 
-  it('should be unable to advance backward by default', () => {
-    const wrapper = mount(Carousel);
+  describe('Autoplay functionality', () => {
+    it.skip('should disable autoplay by default', () => {});
+
+    it.skip('should set the autoplay direction to forward by default', () => {});
+
+    it.skip('should set the autoplay hover pause to true by default', () => {});
+
+    it.skip('should set the autoplay timeout to 2000 by default', () => {});
+
+    it('should begin autoplaying when option specified', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          perPage: 1,
+          autoplay: true,
+          autoplayHoverPause: true
+        },
+        slots: {
+          default: [Slide, Slide]
+        }
+      });
+
+      expect(wrapper.vm.autoplayInterval).toBeDefined();
+      wrapper.vm.pauseAutoplay();
+      expect(wrapper.vm.autoplayInterval).toBe(undefined);
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
+    });
+
+    // start on first slide and allow the autoplay to advance to the second slide
+    it.skip('should display the second slide after waiting the timeout duration with direction forward', () => {});
```

Comment:
These seem more like functional tests to me. I'd like to build something to do UI tests, so maybe we can move these it blocks into a to-do list for functional tests?

Reply:
Moved to a file so we can keep track of them